pyocd/target/family/target_lpc5500.py

```diff
@@ -85,54 +103,63 @@ def _modify_ap1(self, seq):
         return seq
 
     def check_locked_state(self, seq):
-        # Attempt to unlock cores if they are locked (flash is empty etc.)
+        """! @brief Attempt to unlock cores if they are locked (flash is empty etc.)"""
+        # The device is not locked if AP#0 was found and is enabled.
+        if (0 in self.aps) and self.aps[0].is_enabled:
+            return
+        
+        # The debugger mailbox should always be present.
         if not DM_AP in self.aps:
-            exists = coresight.ap.AccessPort.probe(self.dp, DM_AP)
-            if not exists:
-                LOG.error("Error: no DMAP exists")
-                return
-            ap_addr = coresight.ap.APv1Address(DM_AP)
-            ap = coresight.ap.AccessPort.create(self.dp, ap_addr)
```

Comment:
Quick note: unless I'm missing some other changes, this doesn't work for me as DM_AP does not get created (`self.aps` is empty).

I'm using the CMSIS-DAP embedded in the LPC55S69-EVK (revision A2) board.

```
(venv3) [ttt@localhost pyocd-mbedmicro]$ python3 -m pyocd reset -t lpc55s69 -m hw
(venv3) [ttt@localhost pyocd-mbedmicro]$ python3 -m pyocd gdbserver -t lpc55s69 
0000464:INFO:board:Target type is lpc55s69
0000557:INFO:dap:DP IDR = 0x6ba02477 (v2 rev6)
0000571:ERROR:target_lpc5500:cannot request debug unlock; no debugger mailbox AP was found
0000571:ERROR:target_lpc5500:AP#0 was not found, unable to create core 0
0000637:CRITICAL:__main__:No cores were discovered!
Traceback (most recent call last):
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/__main__.py", line 398, in run
    self._COMMANDS[self._args.cmd](self)
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/__main__.py", line 742, in do_gdbserver
    with session:
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/core/session.py", line 343, in __enter__
    self.open()
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/core/session.py", line 461, in open
    self._board.init()
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/board/board.py", line 85, in init
    self.target.init()
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/core/soc_target.py", line 120, in init
    seq.invoke()
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/utility/sequencer.py", line 214, in invoke
    resultSequence = call()
  File "/home/ttt/contribs/pyocd/pyocd-mbedmicro/pyocd/coresight/coresight_target.py", line 220, in check_for_cores
    raise exceptions.DebugError("No cores were discovered!")
pyocd.core.exceptions.DebugError: No cores were discovered!
```

Reply:
That's strange, I wonder why the AHB-APs are completely invisible on your device but on mine they are visible but with `CSW.DEVICEEN` set to 0. My LPC55S69 is the controller on one of the new NXP MCU-Link probes (though it's a prototype build of the probe from several months ago, not brand new).

I'll handle this by adding something else on my todo list: control over the number of not-present APs found in a row that cause the AP scan to stop. Right now it's either a fixed 2 unless `scan_all_aps` is enabled. The LPC5500 family code will set this number to 3 so the DM-AP is always found.

Reply:
Cool, I didn't know the debug probe had that chip. 
The chip on the EVK is a 100pin QFP and the date code says 19271B, so it is a 1B revision chip.

Having the not present APs configurable would help reduce the number of options to set on the command line. Thanks!